### PR TITLE
[PLA-489] add valuation service

### PR DESCRIPTION
**Pull request by Asfiroth**
# Proposed Changes
- Add valuation call

### CODE REVIEW

File: charts/devices-api/values.yaml [hunk 1]
```diff
@@ -78,6 +78,11 @@ env:
   CHATGPT_URL: https://api.openai.com/v1/chat/completions
   AFTERMARKET_DEVICE_CONTRACT_ADDRESS: '0xcf9af64522162da85164a714c23a7705e6e466b3'
   DIMO_CONTRACT_APIURL: https://contracts-api.dev.dimo.zone/
+  NATS_URL: nats-dev:4222
```
Comment (JamesReate) [inline charts/devices-api/values.yaml:81]: need to also have this in `values-prod.yaml` but nats-prod

File: internal/controllers/user_devices_controller.go [hunk 2]
```diff
@@ -510,6 +536,16 @@ func (udc *UserDevicesController) RegisterDeviceForUserFromVIN(c *fiber.Ctx) err
 		}
 	}
 
+	if udc.Settings.IsProduction() {
+		pubAck, err := udc.NATSSvc.JetStream.Publish(udc.NATSSvc.JetStreamSubject, []byte(vin))
+
+		if err != nil {
+			udc.log.Err(err).Msg("failed to publish to NATS")
+		} else {
+			udc.log.Info().Str("vin", vin).Str("user_id", userID).Msgf("published to NATS with ack: %+v", pubAck)
```
Comment (JamesReate) [inline internal/controllers/user_devices_controller.go:545]: update msg to be more specific, eg. `published valuation request to NATS with ack`

File: internal/controllers/user_devices_controller.go [hunk 3]
```diff
@@ -658,6 +694,17 @@ func (udc *UserDevicesController) RegisterDeviceForUserFromSmartcar(c *fiber.Ctx
...
 		return err
 	}
+
+	if udc.Settings.IsProduction() {
+		pubAck, err := udc.NATSSvc.JetStream.Publish(udc.NATSSvc.JetStreamSubject, []byte(vin))
+
+		if err != nil {
+			localLog.Err(err).Msg("Failed to publish to NATS.")
+		} else {
+			localLog.Info().Msgf("Published to NATS with Ack: %+v", pubAck)
```
Comment (JamesReate) [inline internal/controllers/user_devices_controller.go:704]: same here

**Comment by elffjs** [comment]
I will probably not have time to look at this. I'm sure it is good.

**Pull request merged**